File: exercises/week02/src/FireSensor.java
public class FireSensor implements Sensor {
    private String location;
    private double batteryPercentage;
    private String sensorType;

    public FireSensor(String location) {
        this.location = location;
        this.batteryPercentage = 100.0;
        this.sensorType = "Fire sensor";
    }

  @Override
  public boolean isTriggered() {
    return Math.random() < 0.05;
  }

  @Override
  public String getLocation() {
    return null;
  }

  @Override
  public String getSensorType() {
    return null;
  }

  @Override
  public double getBatteryPercentage() {
    return -1;
  }
}

File: exercises/week02/src/FireSensor.scala
import scala.util.Random

class FireSensor(location: String) extends HazardSensor with BatteryPowered {
  private val sensorType: String = "Fire"
  private var batteryPercentage: Double = 100.0

  override def isTriggered: Boolean = {
    if (getBatteryPercentage != 0) batteryPercentage -= 10
    Random.nextDouble() < 0.05
  }

  override def getLocation: String = location

  override def getSensorType: String = sensorType

  override def getBatteryPercentage: Double = batteryPercentage
}
